### PR TITLE
Convert Geode test from Groovy to Java

### PR DESCRIPTION
**Pull request by niteshs7**
Related to #7195 

Converts Geode test from Groovy to Java.

### CODE REVIEW

File: instrumentation/geode-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/geode/PutGetTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+              region.put(key, value);
+              return region.existsValue("SELECT * FROM /test-region");
+            });
+    assertTrue(cacheValue);
+    assertGeodeTrace("existsValue", "SELECT * FROM /test-region");
+  }
+
+  @Test
+  void shouldSanitizeGeodeQuery() throws QueryException {
+
```
Comment (laurit) [inline instrumentation/geode-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/geode/PutGetTest.java:112]:
```suggestion

```

**Review by laurit — APPROVED** [review]

File: instrumentation/geode-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/geode/PutGetTest.java (hunk 1 above)
Comment (niteshs7) [reply instrumentation/geode-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/geode/PutGetTest.java]: Done.

File: instrumentation/geode-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/geode/PutGetTest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                        span.hasName(verb.concat(" test-region"))
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttributesSatisfying(
+                                attributes -> {
+                                  assertThat(attributes)
+                                      .containsEntry(SemanticAttributes.DB_SYSTEM, "geode");
+                                  assertThat(attributes)
+                                      .containsEntry(SemanticAttributes.DB_NAME, "test-region");
+                                  assertThat(attributes)
+                                      .containsEntry(SemanticAttributes.DB_OPERATION, verb);
```
Comment (mateuszrzeszutek) [inline instrumentation/geode-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/geode/PutGetTest.java:158]: Nit: you can use the fluent interface and avoid calling `assertThat(attributes)` multiple times
```suggestion
                                  assertThat(attributes)
                                      .containsEntry(SemanticAttributes.DB_SYSTEM, "geode")
                                      .containsEntry(SemanticAttributes.DB_NAME, "test-region")
                                      .containsEntry(SemanticAttributes.DB_OPERATION, verb);
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by mateuszrzeszutek** [comment]
Thanks @niteshs7 !

**Pull request merged**